extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxProducer.java

```diff
@@ -34,25 +33,22 @@
 
     private static final Logger LOGGER = Logger.getLogger(VertxProducer.class);
 
-    @Inject
-    Vertx vertx;
-
     @Singleton
     @Produces
-    public EventBus eventbus() {
+    public EventBus eventbus(Vertx vertx) {
```

Comment:
So what happens if a user code injects `Vertx`?

Reply:
It's undeployVerticles that is the problem, if it was in a different class these would be fine.

Reply:
OK.